gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +212,744 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+    output = dict(t.aggregate(hl.struct(**metrics_missing)))
+
+    n_fail = 0
+    for metric, n_missing in output.items():
+        if n_missing / n_sites > missingness_threshold:
+            logger.info(
+                "FAILED missingness check for %s: %d sites or %.2f%% missing",
+                metric,
+                n_missing,
+                (100 * n_missing / n_sites),
+            )
+            n_fail += 1
+        else:
+            logger.info(
+                "Passed missingness check for %s: %d sites or %.2f%% missing",
+                metric,
+                n_missing,
+                (100 * n_missing / n_sites),
+            )
+    logger.info("%d missing metrics checks failed", n_fail)
+
+
+def vcf_field_check(
+    t: Union[hl.MatrixTable, hl.Table],
+    header_dict: Dict[str, Dict[str, Dict[str, str]]],
+    row_annotations: List[str],
+    hists: List[str] = HISTS,
+) -> bool:
+    """
+    Check that all VCF fields and descriptions are present in input Table and VCF header dictionary.
+
+    :param t: Input MatrixTable or Tableto be exported to VCF.
+    :param header_dict: VCF header dictionary.
+    :param row_annotations: List of row annotations in MatrixTable.
+    :param hists: List of variant histogram annotations. Default is HISTS.
+    :return: Boolean with whether all expected fields and descriptions are present.
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    # Confirm all VCF fields/descriptions are present before exporting
+    hist_fields = []
+    for hist in hists:
+        hist_fields.extend(
+            [
+                f"{hist}_bin_freq",
+                f"{hist}_n_smaller",
+                f"{hist}_n_larger",
+                f"{hist}_raw_bin_freq",
+                f"{hist}_raw_n_smaller",
+                f"{hist}_raw_n_larger",
+            ]
+        )
+
+    missing_fields = []
+    missing_descriptions = []
+    for item in ["info", "filter"]:
+        if item == "info":
+            annots = row_annotations
+        else:
+            annot_t = t.explode(t.filters)
+            annots = list(annot_t.aggregate(hl.agg.collect_as_set(annot_t.filters)))
+
+        temp_missing_fields = []
+        temp_missing_descriptions = []
+        for field in annots:
+            try:
+                description = header_dict[item][field]
+                if len(description) == 0:
+                    logger.warning(
+                        "%s in T info field has empty description in VCF header!", field
+                    )
+                    temp_missing_descriptions.append(field)
+            except KeyError:
+                logger.warning(
+                    "%s in T info field does not exist in VCF header!", field
+                )
+                # NOTE: some hists are not exported, so ignoring here
+                # END entry is also not exported (removed during densify)
+                if (field not in hist_fields) and (field != "END"):
```

Comment:
`field != "END"` only needs to be checked if the input is a MT, right? also hopefully the `(field not in hist_fields)` check can be removed